### ISSUE TITLE
[Security] Block certain gp commands when sharing a workspace

### ISSUE THREAD

**Issue by shaal** [gitpod-io/gitpod#6782]
## Is your feature request related to a problem? Please describe

Sharing a workspace, comes with a huge security flaw, because you also share your git credentials, and secrets stored in your user account.

## Describe the behaviour you'd like

When a workspace is shared, the commands `gp env` (access to user variables) and `gp credential-helper` (access to git credentials) should be disabled.
That way sharing a workspace with others becomes safe! Users I share the workspace with, can make changes in that specific workspace (that's what I specifically allowed when clicking "Share Workspace"), but they don't have access to my permissions or secrets. They can no longer access my other repos or information that I don't want to share.

## Describe alternatives you've considered

My poor man's hack that I use until this feature is officially implemented, a script that removes the `credential.helper` line from ~/.gitconfig https://github.com/shaal/DrupalPod/pull/46/files#diff-6ed300724c7e75fd4724d06dd9f6d386c13c69e35037f3807aea78811235bafaR6

**Comment by loujaybee** [comment]
Could end up as an IDE / `gp CLI` fix, or a workspace / global fix.

**Comment by akosyakov** [comment]
I don't think it should be fixed on level of GP cli, there are other way to fetch this data from workspace. We should rather reconsider how we do live collaboration in general.

cc @csweichel

**Comment by csweichel** [comment]
> We should rather reconsider how we do live collaboration in general.

💯 agreed. The current sharing concept is much like handing over your laptop, and it would be rather hard adding restrictions to that.

**Comment by shaal** [comment]
The VSCode extension LiveShare  (I wish Gitpod would create an open-source alternative to that), differentiate the type of sharing you can do with others -
* Read Only
* Write Access
* Terminal

**Comment by akosyakov** [comment]
@shaal one day 😄

**Comment by shaal** [comment]
please add `meta: never-stale` label to this issue
